### PR TITLE
Add "Avoid discarding errors during error conversions" example

### LINKED ISSUE #215: add "Avoid discarding errors during error conversions" example
A slightly more complex example than https://github.com/brson/rust-cookbook/issues/214 showing aggregation of custom and crate errors that might use chain_err

Example should show off error-chain crate

cc https://github.com/brson/rust-cookbook/issues/184

### PR DESCRIPTION
**Pull request by ludwigpacifici**
fixes #215

### CODE REVIEW

**Comment by ludwigpacifici** [comment]
Hello @budziq 

To me, the use case (get a random number via HTTP) is not very useful, but it shows the error handling with a relatively short code and covers what was discussed in #215.

If you have some ideas to improve it, let me know.

File: src/basics.md [hunk 1]
```diff
@@ -1144,6 +1145,74 @@ fn main() {
...
+
+[Matching] on different error types returned by a function is possible
+and relatively compact with [error-chain] crate. To do so,
+[`ErrorKind`] is used to determine the error type.
+
+To illustrate the error matching, a random integer generator web
+service will be queried via [reqwest] and then the response will be
+parsed. Errors can be generated by the Rust standard library,
+[reqwest] and by the web service. [`foreign_links`] are used for well
+defined Rust errors. The `errors` macro is used to create an
```
Comment (budziq) [inline src/basics.md]: `errors` is actually not a macro but a clause inside a `error_chain!` macro

File: src/basics.md [hunk 2]
```diff
@@ -1144,6 +1145,74 @@ fn main() {
...
+
+    println!("a random number between 0 and 10: {}", random_value);
+
+    Ok(())
+}
+
+fn main() {
+    if let Err(error) = run() {
+        match *error.kind() {
+            ErrorKind::Io(_) => println!("{:?}", error),
```
Comment (budziq) [inline src/basics.md]: We might want to differentiate the `println!`s here a little. 1-2 word preamble should be enough

File: src/basics.md (hunk 1 above)
Comment (ludwigpacifici) [reply src/basics.md]: Thanks for the clarification! I did not knew that. Is it called a [DSL](https://rustbyexample.com/macros/dsl.html#domain-specific-languages-dsls)?

Comment (budziq) [reply src/basics.md]: > Is it called a DSL?

Yep, the error-chain crate defines a DSL via its macros.

**Comment by ludwigpacifici** [comment]
Thanks for the comments. I updated the PR.

**Comment by budziq** [comment]
Nicely done!

**Comment by ludwigpacifici** [comment]
Thank you @budziq !

**Pull request merged**